src/crimson/os/futurized_store.h

```diff
@@ -60,6 +66,50 @@ class FuturizedStore {
     uint64_t offset,
     size_t len,
     uint32_t op_flags = 0) = 0;
+
+  virtual seastar::future<CollectionRef> open_collection(const coll_t& cid) = 0;
+
+  virtual seastar::future<CollectionRef> create_new_collection(const coll_t& cid) = 0;
+
+  virtual uuid_d get_fsid() const  = 0;
+
+  virtual seastar::future<> write_meta(const std::string& key,
+				       const std::string& value) = 0;
+
+  virtual seastar::future<> do_transaction(
+    CollectionRef ch,
+    ceph::os::Transaction&& txn) = 0;
+
+  virtual FuturizedShardStore& get_local_shard_store() = 0;
+
+  virtual seastar::future<std::tuple<int, std::string>> read_meta(
+    const std::string& key) = 0;
+
+};
+
+class FuturizedShardStore {
+public:
+  FuturizedShardStore() = default;
+  virtual ~FuturizedShardStore() = default;
+  // no copying
+  explicit FuturizedShardStore(const FuturizedShardStore& o) = delete;
+  const FuturizedShardStore& operator=(const FuturizedShardStore& o) = delete;
+
+  // public interfaces called by each shard osd
+  using mount_ertr = FuturizedStore::mount_ertr;
+  virtual mount_ertr::future<> mount() = 0;
+
+  virtual seastar::future<> umount() = 0;
```

Comment:
I meant to drop mount()/umount(), because it looks there's no user from OSD, and it's better to expose them from `FuturizedStore` only as global operations.

Reply:
FuturizedStore will call FuturizedShardStore mount/umount,  so what do you mean drop them?

Reply:
Why cannot drop the mount/umount interfaces from FuturizedShardStore? I don't see there are any external users (e.g. OSD), or am I missing anything?

Reply:
And I don't see FuturizedStore is calling mount/umount at the FuturizedShardStore level.

Reply:
I understand what you mean. will remove them from FuturizedShardStore.  only define them in SeaShardStore and CyanShardStore.